File: liemipeli-test/src/test/kotlin/com/antero/liemipeli/domain/paikat/VarastoTest.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.antero.liemipeli.domain.paikat;

import com.antero.liemipeli.domain.Tuote;
import com.antero.liemipeli.domain.TuoteImpl;
import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.*;

/**
 *
 * @author Antero Oikkonen
 */
public class VarastoTest {

    private Varasto varasto;

    public VarastoTest() {
    }

    @Before
    public void setUp() {
        varasto = new Varasto(1);
    }

    @Test
    public void testHaeTuote() {
        String nimi = "jäkälä";
        varasto.lisaaTuote(new TuoteImpl(nimi, 10, null));
        Tuote t = varasto.haeTuote(nimi);
        assertTrue(t.getNimi().equals(nimi));
    }

    @Test
    public void testHaeTuoteJotaeiole() {
        assertNull(varasto.haeTuote("Musketti"));
    }

    @Test
    public void testLisaaTuoteTayteenVarastoon() {
        assertTrue(varasto.lisaaTuote(new TuoteImpl("yritti", 10, null)));
        assertFalse(varasto.lisaaTuote(new TuoteImpl("jäkälä", 10, null)));
    }

    @Test(expected = IllegalArgumentException.class)
    public void testAsetaVarastonKokoNegatiivinen() {
        varasto.asetaVarastonKoko(-10);
    }

    @Test
    public void testKokoaEivoiasettaaPienemmäksiKuinVarastosaldo() {
        varasto.asetaVarastonKoko(10);
        varasto.lisaaTuote(new TuoteImpl("sokeri", 10, null));
        varasto.lisaaTuote(new TuoteImpl("vehnä", 2, null));
        varasto.lisaaTuote(new TuoteImpl("maito", 10, null));
        assertEquals(3, varasto.asetaVarastonKoko(2));

    }

    @Test
    public void testVarastonArvo() {
        varasto.asetaVarastonKoko(5);
        varasto.lisaaTuote(new TuoteImpl("Vehna", 25, null));
        varasto.lisaaTuote(new TuoteImpl("sokeri", 15, null));
        varasto.lisaaTuote(new TuoteImpl("ruuti", 100, null));

        assertEquals(140, varasto.laskeVarastonArvo());

    }

    @Test
    public void testVarastonArvoHaeTuote() {
        varasto.asetaVarastonKoko(5);
        varasto.lisaaTuote(new TuoteImpl("Vehna", 25, null));
        varasto.lisaaTuote(new TuoteImpl("sokeri", 15, null));
        varasto.lisaaTuote(new TuoteImpl("ruuti", 100, null));
        Tuote t = varasto.haeTuote("ruuti");
        assertEquals("ruuti", t.getNimi());
        assertEquals(40, varasto.laskeVarastonArvo());
        assertEquals(100, t.getHinta());

    }

    @Test(expected = IllegalArgumentException.class)
    public void testAsetaVarastonKokoNegatiiviseksi() {
        varasto.asetaVarastonKoko(-10);
    }

    @Test
    public void asetaVarastonKoko100() {
        assertEquals(100, varasto.asetaVarastonKoko(100));
    }

    @Test
    public void testVarastoSaldo() {
        varasto.asetaVarastonKoko(10);

        varasto.lisaaTuote(new TuoteImpl("miekka", 1000, null));
        varasto.lisaaTuote(new TuoteImpl("haarniska", 2000, null));
        varasto.lisaaTuote(new TuoteImpl("kypärä", 1000, null));

        assertEquals(3, varasto.varastoSaldo());

        Tuote t = varasto.haeTuote("miekka");

        assertEquals(2, varasto.varastoSaldo());
    }

}

File: liemipeli-kotlin/src/test/kotlin/com/antero/liemipeli/domain/paikat/VarastoTest.kt
package com.antero.liemipeli.domain.paikat

import com.antero.liemipeli.domain.TuoteImpl
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import java.util.*


/**
 *
 * @author Antero Oikkonen
 */
class VarastoTest {

    lateinit var varasto: Varasto

    @BeforeEach
    fun setUp() {
        varasto = Varasto(1)
    }

    @Test
    fun testHaeTuote() {
        val nimi = "jäkälä"
        varasto.lisaaTuote(TuoteImpl(nimi, 10, Properties()))
        val t = varasto.haeTuote(nimi)
        assertTrue(t?.nimi == nimi)
    }

    @Test
    fun testHaeTuoteJotaeiole() {
        assertNull(varasto.haeTuote("Musketti"))
    }

    @Test
    fun testLisaaTuoteTayteenVarastoon() {
        assertTrue(varasto.lisaaTuote(TuoteImpl("yritti", 10, Properties())))
        assertFalse(varasto.lisaaTuote(TuoteImpl("jäkälä", 10, Properties())))
    }

    @Test
    fun testKokoaEivoiasettaaPienemmaksiKuinVarastosaldo() {
        varasto.asetaVarastonKoko(10)
        varasto.lisaaTuote(TuoteImpl("sokeri", 10, Properties()))
        varasto.lisaaTuote(TuoteImpl("vehnä", 2, Properties()))
        varasto.lisaaTuote(TuoteImpl("maito", 10, Properties()))
        assertEquals(3, varasto.asetaVarastonKoko(2))

    }

    @org.junit.jupiter.api.Test
    fun testVarastonArvo() {
        varasto.asetaVarastonKoko(5)
        varasto.lisaaTuote(TuoteImpl("Vehna", 25, Properties()))
        varasto.lisaaTuote(TuoteImpl("sokeri", 15, Properties()))
        varasto.lisaaTuote(TuoteImpl("ruuti", 100, Properties()))

        assertEquals(140, varasto.laskeVarastonArvo())

    }

    @Test
    fun testVarastonArvoHaeTuote() {
        varasto.asetaVarastonKoko(5)
        varasto.lisaaTuote(TuoteImpl("Vehna", 25, Properties()))
        varasto.lisaaTuote(TuoteImpl("sokeri", 15, Properties()))
        varasto.lisaaTuote(TuoteImpl("ruuti", 100, Properties()))
        val t = varasto.haeTuote("ruuti")
        assertEquals("ruuti", t?.nimi)
        assertEquals(40, varasto.laskeVarastonArvo())
        assertEquals(100, t?.hinta
        )

    }

    @Test
    fun testAsetaVarastonKokoNegatiiviseksi() {

        val exception = assertThrows(IllegalArgumentException::class.java) {
            varasto.asetaVarastonKoko(-10) }
        assertEquals("koko pitää olla positiivinen", exception.message)
    }

    @Test
    fun asetaVarastonKoko100() {
        assertEquals(100, varasto.asetaVarastonKoko(100))
    }

    @Test
    fun testVarastoSaldo() {
        varasto.asetaVarastonKoko(10)

        varasto.lisaaTuote(TuoteImpl("miekka", 1000, Properties()))
        varasto.lisaaTuote(TuoteImpl("haarniska", 2000, Properties()))
        varasto.lisaaTuote(TuoteImpl("kypärä", 1000, Properties()))

        assertEquals(3, varasto.varastoSaldo())

        varasto.haeTuote("miekka")

        assertEquals(2, varasto.varastoSaldo())
    }

}
